mac/FreeChat/FreeChatAppDelegate.swift

```diff
@@ -0,0 +1,30 @@
+//
+//  FreeChatAppDelegate.swift
+//  FreeChat
+//
+
+import SwiftUI
+
+class FreeChatAppDelegate: NSObject, NSApplicationDelegate, ObservableObject {
+  @Environment(\.openWindow) var openWindow
+  @AppStorage("selectedModelId") private var selectedModelId: String?
+
+  func application(_ application: NSApplication, open urls: [URL]) {
+    let viewContext = PersistenceController.shared.container.viewContext
+    do {
+      let req = Model.fetchRequest()
+      req.predicate = NSPredicate(format: "name IN %@", urls.map({ $0.absoluteString }))
+      let existingModels = try viewContext.fetch(req).compactMap({ $0.url })
+
+      for url in urls {
+        guard !existingModels.contains(url) else { continue }
+        let insertedModel = try Model.create(context: viewContext, fileURL: url)
+        selectedModelId = insertedModel.id?.uuidString
```

Comment:
Not sure why but this is not changing the selected model in the dropdown (see video).

Reply:
Maybe because `selectedModelId` is being initialized with a default value. I addressed this and more issues that I found in https://github.com/psugihara/FreeChat/pull/47